spec/services/supergroup_sections_spec.rb

```diff
@@ -1,26 +1,30 @@
-require "test_helper"
-
 describe SupergroupSections::Sections do
   include SearchApiHelpers
   include TaxonHelpers
 
   let(:taxon_id) { "12345" }
   let(:base_path) { "/base/path" }
   let(:supergroup_sections) { SupergroupSections::Sections.new(taxon_id, base_path) }
+  let(:sections) { supergroup_sections.sections }
 
   describe "#sections" do
-    before(:all) do
-      Supergroups::PolicyAndEngagement.any_instance.stubs(:tagged_content).with(taxon_id).returns(section_tagged_content_list("case_study"))
-      Supergroups::Services.any_instance.stubs(:tagged_content).with(taxon_id).returns(section_tagged_content_list("form"))
-      Supergroups::GuidanceAndRegulation.any_instance.stubs(:tagged_content).with(taxon_id).returns(section_tagged_content_list("guide"))
-      Supergroups::NewsAndCommunications.any_instance.stubs(:tagged_content).with(taxon_id).returns([])
-      Supergroups::Transparency.any_instance.stubs(:tagged_content).with(taxon_id).returns([])
-      Supergroups::ResearchAndStatistics.any_instance.stubs(:tagged_content).with(taxon_id).returns([])
-      @sections = supergroup_sections.sections
+    before do
+      stub_config = {
+        Supergroups::PolicyAndEngagement => section_tagged_content_list("case_study"),
+        Supergroups::Services => section_tagged_content_list("form"),
+        Supergroups::GuidanceAndRegulation => section_tagged_content_list("guide"),
+        Supergroups::NewsAndCommunications => [],
+        Supergroups::Transparency => [],
+        Supergroups::ResearchAndStatistics => [],
+      }
+
+      stub_config.each do |klass, results|
+        allow_any_instance_of(klass).to receive(:tagged_content).and_return(results)
```

Comment:
It's missing off a `.with(taxon_id)` bit. Is that needed?

Reply:
Seems legit 👍